openhands/runtime/impl/eventstream/eventstream_runtime.py

```diff
@@ -466,10 +456,14 @@ def run_action(self, action: Action) -> Observation:
                 return NullObservation('')
             action_type = action.action  # type: ignore[attr-defined]
             if action_type not in ACTION_TYPE_TO_CLASS:
-                return FatalErrorObservation(f'Action {action_type} does not exist.')
+                return ErrorObservation(
```

Comment:
I think we need to return 400 here? This should fail hard because it should be a code implementation error - wrong action type should never be sent here?

Reply:
This part is client-side,  but yeah I get your point